From: =?utf-8?b?0KDRg9GB0LvQsNC9INCY0LbQsdGD0LvQsNGC0L7Qsg==?=
 <lrn1986@gmail.com>
Date: Sun, 10 Jun 2018 21:05:30 +0000
Subject: Use correct stat struct for ftw()

check_dir_mtime() is called by ftw() and is given
the real stat struct, not its glib version (which may
or may not be the same as "struct stat").

This is irrelevant for MSVC (it has no ftw()) and
works correctly for MinGW-w64 (which declares stat
structures correctly). If mingw.org complains, add
a special ifdef for it later.

Origin: upstream, 3.24.39, commit:7e17693cced0db855bfe0997f26633ca8bdd48cc
---
 gtk/updateiconcache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/updateiconcache.c b/gtk/updateiconcache.c
index 787e718..61c42c7 100644
--- a/gtk/updateiconcache.c
+++ b/gtk/updateiconcache.c
@@ -75,7 +75,7 @@ static GStatBuf cache_dir_stat;
 static gboolean cache_up_to_date;
 
 static int check_dir_mtime (const char        *dir,
-                            const GStatBuf    *sb,
+                            const struct stat *sb,
                             int                tf)
 {
   if (tf != FTW_NS && sb->st_mtime > cache_dir_stat.st_mtime)
